sched/misc/assert.c

```diff
@@ -508,7 +508,7 @@ static void dump_deadlock(void)
 void _assert(FAR const char *filename, int linenum,
              FAR const char *msg, FAR void *regs)
 {
-  FAR struct tcb_s *rtcb = running_task();
+  FAR struct tcb_s *rtcb = this_task();
```

Comment:
when assert happen in the interrupt handler, this_task will point to the next running task, not the interrupted task, but mast people expect to dump the later in this case.

Reply:
There may be something that I don't quite undertand, but I changed this just because in case of any crash due to exception, I am getting dump of some irrelevant task, and not the one that causes the crash:

#define running_task() \
  (up_interrupt_context() ? g_running_tasks[this_cpu()] : this_task())

g_running_tasks[this_cpu()] points to the next running task, and this_task points to the interrupted (current) task.

In case of e.g. page fault, the current code will (potentially) dump out information on some other task, and not the one that caused the fault....

Reply:
> There may be something that I don't quite understand, but I changed this just because in case of any crash due to exception, I am getting dump of some irrelevant task, and not the one that causes the crash:
> 
> #define running_task() (up_interrupt_context() ? g_running_tasks[this_cpu()] : this_task())
> 
> g_running_tasks[this_cpu()] points to the next running task, and this_task points to the interrupted (current) task.
> 
> In case of e.g. page fault, the current code will (potentially) dump out information on some other task, and not the one that caused the fault....

no, g_running_tasks point to the interrupted task, this_task may point to the next running task. For example, if the irq handler wake up a new thread, this_task will point to it immediately. g_running_tasks is added to fix this problem.

Reply:
Ir clearly doesn't work as you explained, at least not for risc-v / kernel mode, where assertions always dump out wrong task information at exception.

I can, however remove this patch from this PR and just keep it in our own own branches! It probably didn't belong to this PR anyway.

Reply:
Maybe, riscv forget to set g_running_task to this_task at the end of exception handle. You could search the place under arch/arm which change g_running_task and compare with riscv and find the difference.

Reply:
Yes, it seems that updating the g_running_task was left out when doing a context switch in syscall for risc-v. I made another PR of this: #11961